message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -139,6 +139,17 @@
                 <param index="7">Empty</param>
             </entry>
 
+            <entry name="MAV_CMD_DO_ARM_CHECK" value="42429">
+                <description>Reply with whether or not you can arm and why</description>
```

Comment:
Current pre-arm/arm checks use the gcs_send_text to give an error on why the copter cannot arm (e.g. https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/arming_checks.cpp#L129). We might actually want to rethink pre-arm checks more deeply though. The current implementation seems to arbitrarily split arm and pre-arm checks and only returns one problem at a time (instead of returning a message with a list of problems for example). Parsing the GCS messages gets you what you need for now, it feels a little unreliable though.